controllers/applicationsnapshotenvironmentbinding_controller.go

```diff
@@ -0,0 +1,241 @@
+	GitToken string
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=applicationsnapshotenvironmentbindings,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=applicationsnapshotenvironmentbindings/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=applicationsnapshotenvironmentbindings/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the ApplicationSnapshotEnvironmentBinding object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
+func (r *ApplicationSnapshotEnvironmentBindingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("ApplicationSnapshotEnvironmentBinding", req.NamespacedName)
+
+	// Fetch the ApplicationSnapshotEnvironmentBinding instance
+	var appSnapshotEnvBinding appstudioshared.ApplicationSnapshotEnvironmentBinding
+	err := r.Get(ctx, req.NamespacedName, &appSnapshotEnvBinding)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	log.Info(fmt.Sprintf("Starting reconcile loop for %v %v", appSnapshotEnvBinding.Name, req.NamespacedName))
+
+	applicationName := appSnapshotEnvBinding.Spec.Application
+	environmentName := appSnapshotEnvBinding.Spec.Environment
+	snapshotName := appSnapshotEnvBinding.Spec.Snapshot
+	components := appSnapshotEnvBinding.Spec.Components
+
+	// Create a temp folder to create the gitops resources in
+	tempDir, err := ioutils.CreateTempPath(appSnapshotEnvBinding.Name, r.AppFS)
+	if err != nil {
+		log.Error(err, "unable to create temp directory for gitops resources due to error")
+		r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+		return ctrl.Result{}, fmt.Errorf("unable to create temp directory for gitops resources due to error: %v", err)
+	}
+
+	// Get the Snapshot CR
+	appSnapshot := appstudioshared.ApplicationSnapshot{}
+	err = r.Get(ctx, types.NamespacedName{Name: snapshotName, Namespace: appSnapshotEnvBinding.Namespace}, &appSnapshot)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get the Application Snapshot %s %v", snapshotName, req.NamespacedName))
+		r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+		return ctrl.Result{}, nil
```

Comment:
We should requeue here (along with any other instances where we're not requeueing in the controller)

Returning err instead of nil will trigger the requeue.